resources/waterheater.rb

```diff
@@ -648,51 +649,110 @@ def self.apply_indirect(model, runner, space, cap, vol, t_set, oncycle_p, offcyc
+        # Some parameters to use
+        tank_volume = object.tankVolume.get
+        deadband = object.deadbandTemperatureDifference
+        # Sensors and actuators related to OS water heater object
+        tank_temp_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Water Heater Tank Temperature")
+        tank_temp_sensor.setName("#{combi_sys_id} Tank Temp")
+        tank_temp_sensor.setKeyName(object.name.to_s)
+        tank_loss_energy_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Water Heater Heat Loss Energy")
+        tank_loss_energy_sensor.setName("#{combi_sys_id} Tank Loss Energy")
+        tank_loss_energy_sensor.setKeyName(object.name.to_s)
+        tank_spt_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Schedule Value")
+        tank_spt_sensor.setName("#{combi_sys_id} Setpoint Temperature")
+        tank_spt_sensor.setKeyName(object.setpointTemperatureSchedule.get.name.to_s)
+        alt_spt_sch = object.indirectAlternateSetpointTemperatureSchedule.get
+        altsch_actuator = OpenStudio::Model::EnergyManagementSystemActuator.new(alt_spt_sch, "Schedule:Constant", "Schedule Value")
+        altsch_actuator.setName("#{combi_sys_id} AltSchedOverride")
+      elsif object.is_a? OpenStudio::Model::PumpVariableSpeed
+        pump_actuator = OpenStudio::Model::EnergyManagementSystemActuator.new(object, "Pump", "Pump Mass Flow Rate")
+        pump_actuator.setName("#{combi_sys_id} Pump MFR")
+      end
+    end
 
-    # Actuators
-    altsch_actuator = OpenStudio::Model::EnergyManagementSystemActuator.new(alternate_stp_sch, "Schedule:Constant", "Schedule Value")
-    altsch_actuator.setName("#{obj_name_indirect} AltSchedOverride")
+    mains_temp_sensor = OpenStudio::Model::EnergyManagementSystemSensor.new(model, "Site Mains Water Temperature")
+    mains_temp_sensor.setName("Mains Temperature")
+    mains_temp_sensor.setKeyName("*")
 
     # Program
     indirect_ctrl_program = OpenStudio::Model::EnergyManagementSystemProgram.new(model)
-    indirect_ctrl_program.setName("#{obj_name_indirect} Source Control")
-    indirect_ctrl_program.addLine("Set WH_use = - #{use_heat_sensor.name} -  #{wh_loss_sensor.name}")
-    indirect_ctrl_program.addLine("Set WH_Volume = #{new_heater.tankVolume.get}")
-    indirect_ctrl_program.addLine("If WH_use > (#{tank_temp_sensor.name} - #{stp_temp_sensor.name} + #{deadband(tank_type)}) * WH_Volume * (@RhoH2O #{tank_temp_sensor.name}) * (@CpHW #{tank_temp_sensor.name})")
+    indirect_ctrl_program.setName("#{combi_sys_id} Source MFR Control")
+    indirect_ctrl_program.addLine("Set Rho = @RhoH2O #{tank_temp_sensor.name}")
+    indirect_ctrl_program.addLine("Set Cp = @CpHW #{tank_temp_sensor.name}")
+    indirect_ctrl_program.addLine("Set Tank_Water_Mass = #{tank_volume} * Rho")
+    indirect_ctrl_program.addLine("Set DeltaT = #{Constants.Combi_system_source_temp} - #{tank_spt_sensor.name}")
```

Comment:
Could you obtain this directly from the HX object and avoid the need for a new constant?

Reply:
I am not sure what's the expected change here. HX is also grabbing number from Constants to set up its schedule value, directly obtain it from constants looks more straightforward in this context.

Reply:
I guess I'm thinking that:
```
indirect_ctrl_program.addLine("Set DeltaT = #{Constants.Combi_system_source_temp} - #{tank_spt_sensor.name}")
```
would become something like:
```
indirect_ctrl_program.addLine("Set DeltaT = #{source_manager.schedule.to_ConstantSchedule.get.value} - #{tank_spt_sensor.name}")
```
or whatever the right syntax is.

Does that make sense?

Reply:
Yes I got this, I mean if I removed the constant item there, do you mean in `apply_indirect` function, it's better to directly hardcode `55` as `hx_stp_sch.value` instead of current:

     hx_temp = Constants.Combi_system_source_temp # degree C
     hx_stp_sch.setValue(hx_temp)

This is another source using the constants, if you realized it and prefer that way, I am comfortable to switch.

Reply:
Correct, I would then directly hardcode 55 as you showed above. Your code isn't wrong of course, but there's a subtle difference. In the future, if anyone ever wrote additional code that subsequently overrode the setpoint value for some circumstance, your EMS code as written would not immediately recognize it. But if the EMS code looks at the object value, it would.